index.js

```diff
@@ -44,14 +44,41 @@ exports.run = function(runner, specs) {
     });
   });
 
+  function guid() {
+    function s4() {
+      return Math.floor((1 + Math.random()) * 0x10000)
+        .toString(16)
+        .substring(1);
+    }
+    return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
+  }
+
   function convertOptionsToCliArguments(options) {
     var cliArguments = ['node', 'cucumberjs'];
+    var config = runner.getConfig();
 
     for (var option in options) {
       var cliArgumentValues = convertOptionValueToCliValues(option, options[option]);
 
       if (Array.isArray(cliArgumentValues)) {
         cliArgumentValues.forEach(function (value) {
+          if (config.capabilities.shardTestFiles) {
+            if (option == 'format') {
+              var parts = value.split(':');
```

Comment:
@maximecaruchet my `format` value is evaluated:

```javascript
var dirname = __dirname.replace(/\w:/, '');
var formatJsonString = 'json:' + dirname + '/../test-reports/cucumber.json';
```

Which would not work well with this change.  ...I suppose I should probably `require ('path')` and evaluate the path properly ;-)

I need to do some further investigation and experimentation with this to see if it will suit my needs.  I'm using `multiCapabilities to run against multiple browsers on Selenium grid:

```javascript
config.multiCapabilities = [
            { browserName: 'chrome' },
            { browserName: 'firefox' },
            { browserName: 'internet explorer', version: 10 },
            { browserName: 'safari' }
        ];
```

I noticed in the source: `var opts = assign({}, config.cucumberOpts, config.capabilities.cucumberOpts);`

...I was actually thinking of updating my config to:

```javascript
config.multiCapabilities = [
            { browserName: 'chrome', format: 'json: + chromeTestReport },
            { browserName: 'firefox'', format: 'json: + firefoxTestReport },
            { browserName: 'internet explorer', version: 10', format: 'json: + ieTestReport },
            { browserName: 'safari'', format: 'json: + safariTestReport }
        ];
```

...which would be completely unaffected by your PR.